include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -800,6 +806,85 @@ namespace VectorTools
+                                 const Quadrature<dim>                                     &quadrature,
+                                 const Function<spacedim, typename VectorType::value_type> &function,
+                                 VectorType                                                &vec_result,
+                                 const bool                                                 enforce_zero_boundary,
+                                 const Quadrature<dim-1>                                   &q_boundary,
+                                 const bool                                                 project_to_boundary_first)
+    {
+      const parallel::Triangulation<dim,spacedim> *parallel_tria =
+        dynamic_cast<const parallel::Triangulation<dim,spacedim>*>(&dof.get_tria());
+      Assert (parallel_tria !=0, ExcNotImplemented());
+      Assert (project_to_boundary_first == false, ExcNotImplemented());
+      Assert (enforce_zero_boundary == false, ExcNotImplemented());
+
+      const MPI_Comm &mpi_communicator = parallel_tria->get_communicator();
+
+      typedef typename VectorType::value_type number;
+      Assert (dof.get_fe().n_components() == function.n_components,
+              ExcDimensionMismatch(dof.get_fe().n_components(),
+                                   function.n_components));
+      Assert (vec_result.size() == dof.n_dofs(),
+              ExcDimensionMismatch (vec_result.size(), dof.n_dofs()));
+      Assert (dof.get_fe().degree == fe_degree,
+              ExcDimensionMismatch(fe_degree, dof.get_fe().degree));
+      Assert (dof.get_fe().n_components() == components,
+              ExcDimensionMismatch(components, dof.get_fe().n_components()));
+
+      // set up mass matrix and right hand side
+      typename MatrixFree<dim,number>::AdditionalData additional_data;
+      additional_data.tasks_parallel_scheme =
+        MatrixFree<dim,number>::AdditionalData::partition_color;
+      additional_data.mpi_communicator = mpi_communicator;
+      additional_data.mapping_update_flags = (update_values | update_JxW_values);
+      MatrixFree<dim, number> matrix_free;
+      matrix_free.reinit (mapping, dof, constraints,
+                          QGauss<1>(fe_degree+1), additional_data);
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, components, number> MatrixType;
+      MatrixType mass_matrix(matrix_free);
+
+      const IndexSet locally_owned_dofs = dof.locally_owned_dofs();
+      typedef LinearAlgebra::distributed::Vector<number> LocalVectorType;
+      LocalVectorType vec, rhs, inhomogeneities;
+      matrix_free.initialize_dof_vector(vec);
+      matrix_free.initialize_dof_vector(rhs);
+      matrix_free.initialize_dof_vector(inhomogeneities);
+      constraints.distribute(inhomogeneities);
+      inhomogeneities*=-1.;
+
+      create_right_hand_side (mapping, dof, quadrature, function, rhs);
+      mass_matrix.vmult_add(rhs, inhomogeneities);
+      constraints.condense(rhs);
```

Comment:
@masterleinad Is it this condense call you're having trouble with? Do you really need this one? vmult_add should take care of the constraints internally because it calls distribute_local_to_global(). You might possibly want to call constraints.set_zero().